Revolution Rotation Calculator.py

```diff
@@ -34,33 +34,30 @@ def AbilityRotation(Permutation, AttackSpeed, Activate_Bleeds, Gain, Start_Adren
     Shards = 0
     Adrenaline = Start_Adrenaline
     # --- Calculations begin here --- #
-    AbilityPath.append('AUTO' + ' D: ' + str(round(Current,1)) + ' T: ' + str(round(Clock, 1)) + ' A: ' + str(Adrenaline))
+    AbilityPath.append(f'AUTO D: {round(Current, 1)} T: {round(Clock, 1)} A: {Adrenaline}')
     Current += 50
     Adrenaline += Auto_Adrenaline
-    if Adrenaline > 100:
-            Adrenaline = 100
-    if Adrenaline < 0:
+    if Adrenaline >= 100:
+        Adrenaline = 100
+        for Ability in UltimateIterator:
+            Ready[Ability] = True
+    elif Adrenaline >= 50:
+        for Ability in ThresholdIterator:
+            Ready[Ability] = True
+    elif Adrenaline < 0:
         Adrenaline = 0
-    if Adrenaline >= 50:
-        for Ability in AbilityType:
-            if AbilityType[Ability] == 'T':
-                Ready[Ability] = True
-            elif Adrenaline == 100:
-                for Ability in AbilityType:
-                    if AbilityType[Ability] == 'U':
-                        Ready[Ability] = True
     Clock += 0.6
     Clock = round(Clock, 1)
     while Clock < Time:
         for ability in Permutation:
-            if Clock < Time:
-                if Ready[ability] == True: # Checks if ability can be used
+            if Clock < Time: #TODO: This is used a lot, is it a necessity?

```

Comment:
@NightShadeI This is used a lot, is it a necessity?

Reply:
From memory, since I implemented this a long time ago I think I was having a lot of trouble with abilities being used after the total time had been exceeded. In fact, a few of these IF statements may be useless - probably was me being pedantic. I'll see if i can get around to checking them soon.